stubs/urllib3/@tests/requirements-stubtest.txt

```diff
@@ -0,0 +1 @@
+PySocks>=1.5.6,<2.0,!=1.5.7
```

Comment:
Does this really need to be so specific? (Honest question -- I haven't checked, and I don't know much about `PySocks`!)

Reply:
https://github.com/urllib3/urllib3/blob/1.26.x/setup.py#L99

Reply:
Thanks for the reference! It'll be annoying if we have to continually update this to stop our CI from failing, but I guess if that's what the runtime requires, we should probably have the same requirement here.

Reply:
Does stubtest support the concept of `extras`? The easy way to solve this would be to load a list of extras to install with the package when doing tests

Reply:
> Does stubtest support the concept of `extras`? The easy way to solve this would be to load a list of extras to install with the package when doing tests

Yeah, that's basically the purpose of the `stubtest-requirements.txt` files. All required dependencies are automatically installed by stubtest prior to it stubtesting; this file is for any additional dependencies that we want to specify (which are generally the optional dependencies a package has, if it has any)

Reply:
...I think I misunderstood what you meant — sorry. A better answer for you is that, no, the script we have in CI to install necessary stubtest requirements and then run stubtest doesn't currently know anything about `extras`. But we could probably improve that script — it's here: https://github.com/python/typeshed/blob/master/tests/stubtest_third_party.py

Reply:
I like that idea, quite a lot :) though I can't speak for the other maintainers.

(A change like that should obviously be made in a separate PR, though!)

Reply:
https://github.com/python/typeshed/pull/8467 submitted

Reply:
Thank you!